packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -148,90 +149,6 @@ export abstract class DatabaseInstanceBase extends Resource implements IDatabase
-    { engineMajorVersion: '9.3', parameterGroupFamily: 'postgres9.3' },
-    { engineMajorVersion: '9.4', parameterGroupFamily: 'postgres9.4' },
-    { engineMajorVersion: '9.5', parameterGroupFamily: 'postgres9.5' },
-    { engineMajorVersion: '9.6', parameterGroupFamily: 'postgres9.6' },
-    { engineMajorVersion: '10', parameterGroupFamily: 'postgres10' },
-    { engineMajorVersion: '11', parameterGroupFamily: 'postgres11' },
-  ]);
-
-  public static readonly SQL_SERVER_EE = new DatabaseInstanceEngine('sqlserver-ee', secretsmanager.SecretRotationApplication.SQLSERVER_ROTATION_SINGLE_USER, secretsmanager.SecretRotationApplication.SQLSERVER_ROTATION_MULTI_USER, [
-    { engineMajorVersion: '11', parameterGroupFamily: 'sqlserver-ee-11.0' },
-    { engineMajorVersion: '12', parameterGroupFamily: 'sqlserver-ee-12.0' },
-    { engineMajorVersion: '13', parameterGroupFamily: 'sqlserver-ee-13.0' },
-    { engineMajorVersion: '14', parameterGroupFamily: 'sqlserver-ee-14.0' },
-  ]);
-
-  public static readonly SQL_SERVER_SE = new DatabaseInstanceEngine('sqlserver-se', secretsmanager.SecretRotationApplication.SQLSERVER_ROTATION_SINGLE_USER, secretsmanager.SecretRotationApplication.SQLSERVER_ROTATION_MULTI_USER, [
-    { engineMajorVersion: '11', parameterGroupFamily: 'sqlserver-se-11.0' },
-    { engineMajorVersion: '12', parameterGroupFamily: 'sqlserver-se-12.0' },
-    { engineMajorVersion: '13', parameterGroupFamily: 'sqlserver-se-13.0' },
-    { engineMajorVersion: '14', parameterGroupFamily: 'sqlserver-se-14.0' },
-  ]);
-
-  public static readonly SQL_SERVER_EX = new DatabaseInstanceEngine('sqlserver-ex', secretsmanager.SecretRotationApplication.SQLSERVER_ROTATION_SINGLE_USER, secretsmanager.SecretRotationApplication.SQLSERVER_ROTATION_MULTI_USER, [
-    { engineMajorVersion: '11', parameterGroupFamily: 'sqlserver-ex-11.0' },
-    { engineMajorVersion: '12', parameterGroupFamily: 'sqlserver-ex-12.0' },
-    { engineMajorVersion: '13', parameterGroupFamily: 'sqlserver-ex-13.0' },
-    { engineMajorVersion: '14', parameterGroupFamily: 'sqlserver-ex-14.0' },
-  ]);
-
-  public static readonly SQL_SERVER_WEB = new DatabaseInstanceEngine('sqlserver-web', secretsmanager.SecretRotationApplication.SQLSERVER_ROTATION_SINGLE_USER, secretsmanager.SecretRotationApplication.SQLSERVER_ROTATION_MULTI_USER, [
-    { engineMajorVersion: '11', parameterGroupFamily: 'sqlserver-web-11.0' },
-    { engineMajorVersion: '12', parameterGroupFamily: 'sqlserver-web-12.0' },
-    { engineMajorVersion: '13', parameterGroupFamily: 'sqlserver-web-13.0' },
-    { engineMajorVersion: '14', parameterGroupFamily: 'sqlserver-web-14.0' },
-  ]);
-  /* tslint:enable max-line-length */
-
-  /** To make it a compile-time error to pass a DatabaseClusterEngine where a DatabaseInstanceEngine is expected. */
-  public readonly isDatabaseInstanceEngine = true;
```

Comment:
> Currently, InstanceEngine extended ClusterEngine,
which meant you could pass InstanceEngine when creating a Cluster,
which would fail at deploy time.

it failed at compile time (not in JS)